rmb-sdk-go/peer/connection.go

```diff
@@ -103,9 +132,26 @@ func (c *InnerConnection) loop(ctx context.Context, con *websocket.Conn, output,
 		case data := <-outputCh:
 			output <- data
 			lastPong = time.Now()
-		case data := <-input:
-			if err := con.WriteMessage(websocket.BinaryMessage, data); err != nil {
-				return err
+		case sent := <-c.writer:
+			// should we remove this flag?
+			c.busy = true
+			defer func() {
+				c.busy = false
+			}()
```

Comment:
You can't use a `defer` in a loop. this is wrong. Read about it

This is why u still had to set c.busy=false later in the code at line (152)

I suggest that u move this entire branch logic to a function so u can use the defer to release the busy flag.

Reply:
As i remember the busy flag was intended to make a quick return if the connection is not established so senders can already try another connection until this one is restored. It means it shouldn't be turned on/off inside the <Send> branch, but instead ONLY when that connection instance is reconnecting to relay.